lib/ui/src/modules/ui/libs/hierarchy.js

```diff
@@ -46,24 +46,47 @@ function fillHierarchy(namespaces, hierarchy, story) {
   fillHierarchy(namespaces.slice(1), childHierarchy, story);
 }
 
-export function createHierarchy(stories) {
-  const hierarchyRoot = {
+export function createHierarchyRoot(name = '') {
+  return {
     isNamespace: true,
     namespaces: [],
-    name: '',
+    name,
     map: new Map(),
   };
+}
+
+export function createHierarchies(stories) {
+  const rootMap = {};
 
   if (stories) {
     stories.forEach(story => {
-      const { namespaces } = story;
+      const { rootName = '', namespaces } = story;
       const name = namespaces[namespaces.length - 1];
+      const hierarchyRoot =
+        rootMap[rootName] || (rootMap[rootName] = createHierarchyRoot(rootName));
 
       fillHierarchy(namespaces, hierarchyRoot, { ...story, name });
     });
   }
 
-  return hierarchyRoot;
+  return Object.keys(rootMap).map(rootName => rootMap[rootName]);
+}
+
+export function resolveStoryRoots(storyName = '', rootSeparator) {
+  if (!rootSeparator) {
+    return ['', storyName];
```

Comment:
I think in this method better return an object, instead of relying on indices in the array.

Reply:
Agreed. Changed it, so it returns an object.

Btw I renamed resolveStoryRoots to resolveStoryHierarchyRoots.